catalog/task/collect-data/0.3/collect-data.yaml

```diff
@@ -51,30 +63,36 @@ spec:
           value: '$(params.releasestrategy)'
         - name: "SNAPSHOT"
           value: '$(params.snapshot)'
-        - name: "SUBDIR"
-          value: '$(workspaces.data.path)/$(params.subdirectory)'
       script: |
         #!/usr/bin/env sh
         set -e
 
         if [ -n "$(params.subdirectory)" ]; then
-          mkdir -p $SUBDIR
+          mkdir -p $(workspaces.data.path)/$(params.subdirectory)
         fi
 
-        cd $SUBDIR
-
-        get-resource "release" "${RELEASE}" > "release.json"
+        RELEASE_PATH=$(params.subdirectory)/release.json
+        echo -n $RELEASE_PATH > $(results.release.path)
+        get-resource "release" "${RELEASE}" > $(workspaces.data.path)/$RELEASE_PATH
 
-        get-resource "releaseplan" "${RELEASE_PLAN}" > "release_plan.json"
+        RELEASEPLAN_PATH=$(params.subdirectory)/release_plan.json
+        echo -n $RELEASEPLAN_PATH > $(results.releasePlan.path)
+        get-resource "releaseplan" "${RELEASE_PLAN}" > $(workspaces.data.path)/$RELEASEPLAN_PATH
 
+        RELEASEPLANADMISSION_PATH=$(params.subdirectory)/release_plan_admission.json
+        echo -n $RELEASEPLANADMISSION_PATH > $(results.releasePlanAdmission.path)
         get-resource "releaseplanadmission" "${RELEASE_PLAN_ADMISSION}" \
-          > "release_plan_admission.json"
+          > $(workspaces.data.path)/$RELEASEPLANADMISSION_PATH
 
-        get-resource "releasestrategy" "${RELEASE_STRATEGY}" > "release_strategy.json"
+        RELEASESTRATEGY_PATH=$(params.subdirectory)/release_strategy.json
+        echo -n $RELEASESTRATEGY_PATH > $(results.releaseStrategy.path)
+        get-resource "releasestrategy" "${RELEASE_STRATEGY}" > $(workspaces.data.path)/$RELEASESTRATEGY_PATH
 
-        get-resource "snapshot" "${SNAPSHOT}" "{.spec}" > "snapshot_spec.json"
+        SNAPSHOTSPEC_PATH=$(params.subdirectory)/snapshot_spec.json
+        echo -n $SNAPSHOTSPEC_PATH > $(results.snapshotSpec.path)
+        get-resource "snapshot" "${SNAPSHOT}" "{.spec}" > $(workspaces.data.path)/$SNAPSHOTSPEC_PATH
 
-        cat "snapshot_spec.json" | jq -cr '.components[0].containerImage' | tr -d "\n" \
+        cat $SNAPSHOTSPEC_PATH | jq -cr '.components[0].containerImage' | tr -d "\n" \
```

Comment:
I believe you need `$(workspaces.data.path)/$SNAPSHOTSPEC_PATH` here.

Reply:
right, good catch